react-components/src/analysis/view/dialogs/AnalysisParametersDialog.js

```diff
@@ -19,54 +21,32 @@ import TableBody from "@material-ui/core/TableBody";
 import { withStyles } from "@material-ui/core/styles";
 
 
-const ArgumentType = {
-    Input: "Input",
-    FileInput: "FileInput",
-    FolderInput: "FolderInput",
-    MultiFileSelector: "MultiFileSelector",
-    FileFolderInput: "FileFolderInput",
-    EnvironmentVariable: "EnvironmentVariable",
-    Flag: "Flag",
-    Info: "Info",
-    MultiLineText: "MultiLineText",
-    Integer: "Integer",
-    Double: "Double",
-    Text: "Text",
-    TextSelection: "TextSelection", // For selecting from a list of string values.
-    IntegerSelection: "IntegerSelection", // For selecting from a list of integers
-    DoubleSelection: "DoubleSelection", // For selecting from a list of doubles
-    TreeSelection: "TreeSelection",
-    Output: "Output",
-    FileOutput: "FileOutput",
-    FolderOutput: "FolderOutput",
-    MultiFileOutput: "MultiFileOutput",
-    ReferenceGenome: "ReferenceGenome",
-    ReferenceSequence: "ReferenceSequence",
-    ReferenceAnnotation: "ReferenceAnnotation",
-    Selection: "Selection",
-    ValueSelection: "ValueSelection",
-    Number: "Number",
-    Group: "Group",
-};
-
 function ParameterValue(props) {
-    const {parameter, classes, diskResourceUtil, onValueClick} = props;
-    let info_type = parameter.info_type;
+    const {
+        parameter,
+        parameter: {
+            info_type,
+            param_type,
+            param_value: {value},
+        },
+        classes,
+        diskResourceUtil,
+        onValueClick,
+    } = props;
     let valid_info_type = (!(info_type === "ReferenceGenome")
         && !(info_type === "ReferenceSequence")
         && !(info_type === "ReferenceAnnotation"));
-    let displayValue = parameter.param_value.value.display ? parameter.param_value.value.display : parameter.param_value.value;
+    let displayValue = value ? (value.display ? value.display : value) : "";
 
     if (!displayValue) {
         displayValue = "";
```

Comment:
This `if` check can be removed now, since it's already handled in the `let` statement above.